apps/cowswap-frontend/src/modules/account/containers/AccountDetails/HwAccountIndexSelector.tsx

```diff
@@ -0,0 +1,34 @@
+import { useAtom } from 'jotai/index'
+import { FormEvent, useCallback, useRef } from 'react'
+
+import styled from 'styled-components/macro'
+
+import { hwAccountIndexAtom } from 'modules/wallet/api/state'
+
+const Wrapper = styled.form`
+  font-size: 14px;
+  margin: 10px 0;
+`
+
+// TODO: add styles
+export function HwAccountIndexSelector() {
+  const accountIndexRef = useRef<HTMLInputElement>(null)
+  const [hwAccountIndex, setHwAccountIndex] = useAtom(hwAccountIndexAtom)
+
+  const onHwAccountIndexChange = useCallback(
+    (event: FormEvent) => {
+      event.preventDefault()
+
+      setHwAccountIndex(+(accountIndexRef.current?.value || 0))
```

Comment:
Question: why use a ref rather than getting the form values?

Not saying you should do the later, just want to understand the reasoning.

Reply:
Good point, thanks! Refactored